### PR TITLE
Edit Post: Hide overflowing content in visual editor wrapper to prevent block popovers from creating scrollbars

### PR DESCRIPTION
**Pull request by andrewserong**
## What?

Add `overflow: hidden` to the `.edit-post-visual-editor` class, so that block popovers don't cause additional scrollbars in the editor canvas.

## Why?

Block popovers are rendered into a popover slot that has the `edit-post-visual-editor` wrapper as its parent, with `position: relative` set on it. The current behaviour with `BlockPopover` and floating UI is that popovers are rendered using absolute positioning, so when a block popover goes past the bottom of the screen, it results in the `edit-post-visual-editor` wrapper overflowing. While we could also look into updating the popover behaviour here, I thought a good step would be to prevent these overflowing popovers from causing a scrollbar to be displayed. (The real post content is scrolled via the `.interface-interface-skeleton__content` wrapper, so I _think_ this should hopefully be a safe change).

This issue currently only appears in the post editor, as the site editor already has a corresponding `overflow: hidden` rule [on this line](https://github.com/WordPress/gutenberg/blob/d27cb3233b1b156d09442ccdf6da8e9fa157940e/packages/edit-site/src/components/block-editor/style.scss#L19), added back in https://github.com/WordPress/gutenberg/issues/44770. To test and reproduce the issue, if you're on a Mac, you'll likely want to set scrollbars in the OS to display Always for the issue to be visible.

The kinds of popovers that result in the additional scrollbar include the block toolbar once it's scrolled past the bottom of the screen, but also Padding and Margin visualisers in the Group block, for example. I noticed this issue while working on a visualiser for position support over in https://github.com/WordPress/gutenberg/pull/49321.

## How?

* Add `overflow: hidden` to the `edit-post-visual-editor` class.

## Testing Instructions

Before testing, make sure your OS is set to always display scrollbars. On a Mac, go to System Preferences > General and set it to Always. Here's a screenshot:

1. On `trunk` with a long post in the post editor, select a block somewhere around the middle of the page and scroll up and down the editor canvas. Notice that a second scrollbar appears once the block toolbar goes past the bottom of the screen.
2. With a Group block that's large enough for the padding or margin to go off the vertical edges of the screen (e.g. add a whole bunch of paragraphs inside it), go to adjust the padding or margins and notice that the visualiser results in an extra scrollbar.
3. Checkout this PR, and repeat the above steps — there shouldn't be an additional scrollbar, and scrolling the document should still work as before.
4. Check that the template editing mode within the post editor still works as expected.

### Before

https://user-images.githubusercontent.com/14988353/233546738-642a93ae-7402-43e3-8a15-105af6725d39.mp4

### After

https://user-images.githubusercontent.com/14988353/233546805-b764296c-6a3c-43b4-88e2-68d3779c86f2.mp4

### CODE REVIEW

**Comment by andrewserong** [comment]
Apologies for the large number of pings for reviews — while this is only a single line change, I thought I'd cast the net wide, just in case anyone's aware of problems this rule might cause! Feel free to ignore if this isn't an area any of you are working in 🙂

**Comment by talldan** [comment]
> I thought I'd cast the net wide, just in case anyone's aware of problems this rule might cause!

From memory, I think the template editing mode and the Mobile / Tablet views are worth testing. I couldn't spot any regressions.

**Comment by ramonjd** [comment]
I could reproduce problem and `overflow: hidden;` fixes it for me too. Strangely I couldn't reproduce in the site editor 🤔

**Comment by andrewserong** [comment]
> Strangely I couldn't reproduce in the site editor 🤔

Thanks for testing! The site editor already has the `overflow: hidden` rule, as it was added for the Browse mode.

**Comment by aaronrobertshaw** [comment]
I could replicate the issue and confirm this PR fixes it for me.

FWIW the scrollbar setting was different for me using MacOS Ventura. It was under Appearance rather than General but the double scrollbars showed up without turning on "always" show scrollbars.

Screenshot 2023-04-21 at 3 40 35 pm

**Comment by ramonjd** [comment]
> Thanks for testing! The site editor already has the overflow: hidden rule, as it was added for the Browse mode.

That would do it 🤕

**Comment by jasmussen** [comment]
Thank you for attention to detail here, this one is important. I recall similar issues in the past, in case https://github.com/WordPress/gutenberg/issues/30055 can provide some context on what used to be the issue and how that was fixed.

**Comment by andrewserong** [comment]
Thanks for linking through that earlier discussion @jasmussen!

It looks like the fix for the site editor was in https://github.com/WordPress/gutenberg/pull/43332 from @ciampo and reviewed/merged by @talldan, which added the `overflow: hidden` rule to the `.edit-site-visual-editor` class. That means this PR is consistent with that fix, but is rolling it out for the post editor now. (I thought the rule was added in #44770 with Browse mode, but it turns out that PR just moved where the rule lived 😄)

I think this one should be good to go now, but I'll just add a slightly bigger ping to @WordPress/gutenberg-core just in case.

**Review by ramonjd — APPROVED** [review]
Based on fix for site editor, this LGTM

**Comment by andrewserong** [comment]
Thanks for the approving review @ramonjd! Out of an abundance of caution, since this updates a styling rule of a prominent part of the editor, I'll leave this PR open until I'm back on Wednesday, and will merge in then if there aren't any objections 🙂 

(If anyone else would like to, feel free to merge in before then)

**Review by aristath — APPROVED** [review]

**Review by apeatling — APPROVED** [review]
✅ Confirmed works as expected LGTM

**Comment by andrewserong** [comment]
Thanks for the confidence check / reviews everyone! 🙇

**Pull request merged**
